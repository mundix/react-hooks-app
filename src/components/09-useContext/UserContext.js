import { createContext } from 'react';
// Esto es un higher order componetn, esto sinifica que peude tener hijos y es como <Component/>
export const UserContext = createContext(null);
