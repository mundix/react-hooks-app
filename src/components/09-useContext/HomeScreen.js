import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    const {user} = useContext(UserContext);
    console.log(user);

    return (
        <>
            <h1>Home Page</h1>
            <hr />
            <pre>
                {JSON.stringify(user, null, 4)}
            </pre>
        </>
    )
}
