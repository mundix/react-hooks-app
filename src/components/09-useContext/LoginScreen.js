import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    // 1 Obtebner referencia del userContext
   const {setUser} = useContext(UserContext);
    // console.log(context);
    // 2 setUser

    // {
    //     id: 123,
    //     name: 'Lo que sea'
    // }

    return (
        <>
             <h1>Login Page</h1>
             <hr />
             <button className="btn btn-primary"
                onClick={()=>{ setUser( {
                    id: 123,
                    name: 'Edmundo'
                })}}
             >
                 Login 
             </button>
        </>
    )
}
