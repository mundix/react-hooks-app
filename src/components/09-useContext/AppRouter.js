// Una app peude tener todos los routers que necesite 
import React from 'react';

import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            {/* Este es el fragment  */}
            <div>

                <NavBar />
                <div className="container">
                    <Switch>
                        <Route exact path='/' component={HomeScreen} />

                        <Route exact path='/about' component={AboutScreen} />
                        <Route exact path='/login' component={LoginScreen} />

                        {/* Estop es un default por si no encuentran nada, esto puede ser un 404 o se puede usar un redirect */}
                        {/* <Router component={ HomeScreen }/> */}
                        {/* Si no encuentra ninguna de estas direccion redirecciona al home  */}
                        <Redirect to='/' />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
