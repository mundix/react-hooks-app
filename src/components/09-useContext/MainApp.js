import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext'; //este un componente useContext
// La idea es que no haga refresh para cambair entre pantallas, par eso es el router 

export const MainApp = () => {
    // const user = {
    //     id: 1234,
    //     name: 'Edmundo',
    //     email: 'ce.pichardo@gmail.com'
    // }

    const [user, setUser] = useState({
        id: 1234
    });

    return (
        // <UserContext.Provider value={user}>
        // REgreso un obnjeto con propiedades , para enviar mas cosas
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <AppRouter />
        </UserContext.Provider>
    )
}
