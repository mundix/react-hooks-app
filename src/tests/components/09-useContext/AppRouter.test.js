import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Prueba en <AppRouter />', () => {

    const user = {
        id: 1,
        email: 'ce.pichardo@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={user}>
            <AppRouter/>
        </UserContext.Provider>
    );

    test('DEbe de mostrarse correcamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    

})
