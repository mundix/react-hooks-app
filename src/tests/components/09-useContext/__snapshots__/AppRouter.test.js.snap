// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <AppRouter /> DEbe de mostrarse correcamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item  nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item  nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        Home Page
      </h1>,
      <hr />,
      <pre />,
    ]
  </div>
</div>
`;
